cluster/addons/fluentd-gcp/fluentd-gcp-configmap.yaml

```diff
@@ -397,13 +412,19 @@ data:
       # Use multiple threads for processing.
       num_threads 2
       use_grpc true
-      labels {
-        # The logging backend will take responsibility for double writing to
-        # the necessary resource types when this label is set.
-        "logging.googleapis.com/k8s_compatibility": "true"
-      }
+      # Use Metadata Agent to get monitored resource.
+      enable_metadata_agent true
     </match>
 
+    # Attach local_resource_id for 'k8s_node' monitored resource.
+    <filter **>
+      @type record_transformer
+      enable_ruby true
+      <record>
+        "logging.googleapis.com/local_resource_id" ${"k8s_node.#{ENV['NODE_NAME']}"}
```

Comment:
Careful, I don't think `NODE_NAME` exists, it would need to be added to the daemonset yaml as an environment variable pulled from the downwards api. This is what I've used historically.

```
        - env:
          - name: NODE_NAME
            valueFrom:
              fieldRef:
                apiVersion: v1
                fieldPath: spec.nodeName
```

Reply:
Oops. This used to be there. I removed it in the last commit thinking it was only used by the `metadata_agent_url` and no longer needed. Just added it back.